battleship.py

```diff
@@ -134,7 +154,22 @@ def addShips(grid, numShips):
 Returns: None
 '''
 def drawGrid(data, canvas, grid, showShips):
-    return
+    cellsize = data["cell"]
+    for row in range(data["rows"]):
+        for col in range(data["cols"]):
+            if grid[row][col] == SHIP_UNCLICKED and showShips == False:
+                canvas.create_rectangle(col*cellsize,row*cellsize,(col+1)*cellsize,(row+1)*cellsize,fill="blue",width=1)
+            else :
+                canvas.create_rectangle(col*cellsize,row*cellsize,(col+1)*cellsize,(row+1)*cellsize,fill="blue",width=1)
+            if grid[row][col] == SHIP_CLICKED:
+                canvas.create_rectangle(col*cellsize,row*cellsize,(col+1)*cellsize,(row+1)*cellsize,fill="red",width=1)
+            elif grid[row][col] == EMPTY_CLICKED:
+                canvas.create_rectangle(col*cellsize,row*cellsize,(col+1)*cellsize,(row+1)*cellsize,fill="white",width=1)

```

Comment:
Ships should be visible on the user board. Condition for SHIP_UNCLICKED and showships is True is missing.